_source/_posts/2022-01-30-local-https-java.adoc

```diff
@@ -0,0 +1,285 @@
+
+Generate another set of certificates with `mkcert`, but this time set the output format to `pkcs12`, which is a format the Java Keystore understands.
+
+[source,bash]
+----
+# create the directory
+mkdir -p ~/.config/spring-boot
+
+# generate a certificate in pkcs12 format
+mkcert -pkcs12 -p12-file ~/.config/spring-boot/local-tls.p12 localhost
+----
+
+Create a properties file `~/.config/spring-boot/spring-boot-devtools.properties` containing the server port and newly generated key store location.
+
+[source,properties]
+----
+# Set the port
+server.port=8443
+
+# configure the key store path
+server.ssl.key-store=${user.home}/.config/spring-boot/local-tls.p12
+----
+
+Now restart your Spring Boot application, and open your browser to `\https://localhost:8443` to access your newly secured application!
+
+[TIP]
+====
+I didn't need to make any changes to my application because it's using https://docs.spring.io/spring-boot/docs/current/reference/html/using.html[Spring Boot DevTools], which automatically includes the above properties file. If you are *not* using DevTools, you can mimic this behavior by setting an environment variable before starting your application:
+
+[source,bash]
+----
+SPRING_CONFIG_IMPORT='${user.home}/.config/spring-boot/spring-boot-devtools.properties'
+----
+====
+
+== Bonus: Use a different hostname
+
+The above examples use `localhost`, which works, but you might want to use a different hostname. One easy way to do this is to add a hostname alias to your `/etc/hosts` file:
+
+[source,bash]
+----
+sudo sh -c 'echo "127.0.0.1 local.example" >> /etc/hosts'
+----
+
+In the above examples, when generating certificates with `mkcert`, replace `localhost` with your domain name, for example:
+
+[source,bash]
+----
+mkcert -key-file devcerts/key.pem -cert-file devcerts/cert.pem local.example
+----
+
+WARNING: Use a domain name that you own or NOT in the official https://data.iana.org/TLD/tlds-alpha-by-domain.txt[top-level domain list] to ensure there are no conflicting or hijacked DNS entries.
+
+== Potential problems using HTTPS for local development
+
+It's not all sunshine and roses, there are a few things to watch out for. If you are a developer that works on multiple applications.
+If you are switching between applications that run on HTTP and HTTPS prototypes, your browser may automatically switch to HTTPS, especially one of your applications sets the https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security[HTTP Strict-Transport-Security (HTST) headers]. If you run into this type of problem you can clear the https://www.a2hosting.com/kb/getting-started-guide/internet-and-networking/clearing-a-web-browsers-ssl-state[SSL State] and/or https://www.thesslstore.com/blog/clear-hsts-settings-chrome-firefox/[HSTS settings] in your browser.
+
+TIP: Use different hostname for each application you are working on to avoid any browser caching.
```

Comment:
```suggestion
TIP: Use a different hostname for each application you are working on to avoid  browser caching.
```
add one word, remove another...